pkg/operator/webhook_configuration.go

```diff
@@ -182,39 +115,48 @@ func applyUnstructured(resourceClient dynamic.ResourceInterface, path string, re
 		return nil, false, err
 	}
 
-	existingCopy := existing.DeepCopy()
-	modified, err := ensureObjectMeta(existingCopy, required)
-	if err != nil {
-		return nil, false, err
+	// Providing upgrade compatibility with service-ca-bundle operator
+	// and ignore clientConfig.caBundle changes on "inject-cabundle" label
+	if required.GetAnnotations() != nil && required.GetAnnotations()[genericCABundleInjectorLabel] != "" {
+		copyValidatingWebhookCABundle(existing, required)
 	}
 
-	if !modified {
-		return existingCopy, false, nil
-	}
+	modified := resourcemerge.BoolPtr(false)
+	existingCopy := existing.DeepCopy()
 
-	requiredSpec, exists, err := unstructured.NestedFieldNoCopy(required.Object, path)
-	if err != nil {
-		return nil, false, err
-	}
-	if !exists {
-		// No spec to update
+	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, required.ObjectMeta)
+	if !*modified && existingCopy.GetGeneration() == expectedGeneration {
 		return existingCopy, false, nil
 	}
-
 	// at this point we know that we're going to perform a write.  We're just trying to get the object correct
 	toWrite := existingCopy // shallow copy so the code reads easier
-	if err := unstructured.SetNestedField(toWrite.Object, requiredSpec, path); err != nil {
+	toWrite.Webhooks = required.Webhooks
+
+	klog.V(4).Infof("ValidatingWebhookConfiguration %q changes: %v", required.GetNamespace()+"/"+required.GetName(), jsonPatchNoError(existing, toWrite))
+
+	actual, err := client.Update(context.TODO(), toWrite, metav1.UpdateOptions{})
+	if err != nil {
 		return nil, false, err
 	}
+	if *modified || actual.GetGeneration() > expectedGeneration {
+		reportUpdateEvent(recorder, required, err)
+		return actual, true, nil
+	}
+	return actual, false, nil
+}
 
-	if klog.V(4) {
-		klog.Infof("%s %q changes: %v", required.GetKind(), required.GetNamespace()+"/"+required.GetName(), jsonPatchNoError(existing, toWrite))
+func copyValidatingWebhookCABundle(existing, required *admissionregistrationv1.ValidatingWebhookConfiguration) {
+	existingMutatingWebhooks := make(map[string]admissionregistrationv1.ValidatingWebhook)
```

Comment:
This should be validating
```suggestion
	existingValidatingWebhooks := make(map[string]admissionregistrationv1.ValidatingWebhook)
```